feat(ast_codegen, span): process `Span` through ast_codegen

overlookmotel opened the pull request:
Add `#[ast]` attr to `Span`.

Due to how AST codegen works, this necessitates putting the type def in a separate file. https://github.com/oxc-project/oxc/pull/4696#issuecomment-2271781995

overlookmotel commented:
* **#4703** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4703?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

rzvxa commented:
It actually wouldn't panic for this specific file, because it doesn't include any other enum/struct definitions. I don't panic on `impl` items, Just on unmarked enum/structs.

But It is better to keep the convention, Even if we want to merge impls and definitions it can be done in one go.

overlookmotel committed:
feat(ast_codegen, span): process `Span` through ast_codegen (#4703)

Add `#[ast]` attr to `Span`.

Due to how AST codegen works, this necessitates putting the type def in a separate file. https://github.com/oxc-project/oxc/pull/4696#issuecomment-2271781995
